packages/apollo/src/hooks.ts

```diff
@@ -11,27 +11,69 @@ import {
   QueryTuple,
   TypedDocumentNode,
 } from "@apollo/client";
+import { notEmpty } from "@uplift-ltd/ts-helpers";
 import { DocumentNode } from "graphql";
 import { GRAPHQL_AUTH_URL, GRAPHQL_UNAUTH_URL } from "./constants";
 
 export interface ExtraOptions {
   auth?: boolean;
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface EnhancedQueryHookOptions<TData = any, TVariables = OperationVariables>
+  extends QueryHookOptions<TData, TVariables> {
+  skipVariables?: (keyof TVariables)[];
+}
+
 export type EnhancedQueryResult<TData, TVariables> = QueryResult<TData, TVariables> & {
+  fetchingMore: boolean;
   initialLoading: boolean;
   refetching: boolean;
-  fetchingMore: boolean;
 };
 
+type DeriveSkipFromSkipAndSkipVariablesArgs<
+  Variables extends OperationVariables = OperationVariables
+> = {
+  skip?: QueryHookOptions["skip"];
+  predicate?: (value: unknown) => boolean;
+  skipVariables?: (keyof Variables)[];
+  variables?: Variables;
+};
+
+/**
+ * Derives whether the query should be skipped based on an explicit skip, or
+ * the existence of query variables that should be notEmpty based on a provided
+ * list of skipVariables (a list of variable keys).
+ *
+ * Predicate defaults to `notEmpty` but can be customized
+ */
+function deriveSkipFromSkipAndSkipVariables<Variables = OperationVariables>({
+  skip,
+  predicate = notEmpty,
+  skipVariables = [],
+  variables,
+}: DeriveSkipFromSkipAndSkipVariablesArgs<Variables>) {
+  // early return if we have explicitly marked it as skip,
+  // or if there are no variables defined
+  if (skip) return skip;
```

Comment:
Should we already return if explicit false is passed in too?
```suggestion
  if (typeof skip === 'boolean') return skip;
```

Reply:
I was thinking that if skip was set to false, then we could still check the skipVariables. But we'd only always skip if it were true

this would allow us to combine a non-variable skip condition with skipVariables

Reply:
Sounds good!